plugins/analytics-module-segment/config.d.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Config {
+  app: {
+    // TODO: Only marked as optional because backstage-cli config:check in the
+    // context of the monorepo is too strict. Ideally, this would be marked as
+    // required.
```

Comment:
I think really think it makes sense to force Segment to be used in all environments just because the module is installed in a project?  I think it's best to keep both the `analytics` and `segment` levels optional in this schema. Probably makes sense to have the implementation blow up early at runtime if config isn't available though.